### PR TITLE
feat: More validation for OpenID credential issuers in attribute sharing

### PR DESCRIPTION
**Pull request by aterga**
# Motivation

Fail early by pre-validating OpenID credential issuers used in attribute sharing scopes.

# Changes

* Adds validation function for OpenID credential issuer identifiers requiring HTTPS, prohibiting query/fragment parameters, and enforcing a maximum length of 1024 characters

# Tests

* Updates all test cases to use valid HTTPS URLs for issuer identifiers
* Adds extensive test coverage for validation edge cases including IPv6 addresses, length limits, and invalid formats

### CODE REVIEW

File: src/internet_identity_interface/src/internet_identity/types/attributes.rs [hunk 1]
```diff
@@ -52,6 +54,58 @@ impl std::fmt::Display for AttributeScope {
     }
 }
 
+fn elipsized(s: &str, max_len: usize) -> String {
+    if s.len() > max_len {
+        format!("{}{}", &s[..max_len - 3], "...")
```
[the remark this answers is not in the document]
Comment (aterga) [reply src/internet_identity_interface/src/internet_identity/types/attributes.rs]: Addressed

File: src/internet_identity_interface/src/internet_identity/types/attributes.rs [hunk 2]
```diff
@@ -52,6 +54,85 @@ impl std::fmt::Display for AttributeScope {
...
+            clamp_string_len(issuer, OPENID_ISSUER_MAX_LENGTH)
+        ));
+    }
+
+    if issuer.contains("#") {
+        problems.push(format!(
+            "Invalid issuer `{}` in attribute scope (must not contain fragment '#' characters)",
+            clamp_string_len(issuer, OPENID_ISSUER_MAX_LENGTH)
+        ));
+    }
```
[the remark this answers is not in the document]
Comment (aterga) [reply src/internet_identity_interface/src/internet_identity/types/attributes.rs:127]: We can use an overly strict check for now, since we don't expect any issues with the supported OIDC provider URLs.

We will eventually switch to parsing the URLs using a library (but we should avoid new dependencies for now)

File: src/internet_identity_interface/src/internet_identity/types/attributes.rs [hunk 3]
```diff
@@ -60,18 +141,21 @@ impl TryFrom<&str> for AttributeScope {
...
                 let issuer = parts
                     .next()
                     .ok_or_else(|| format!("Missing issuer in attribute scope: {}", value))?
                     .to_string();
-                if issuer.is_empty() {
-                    return Err(format!("Missing issuer in attribute scope: {}", value));
-                }
+
+                validate_openid_credential_issuer_identifier(&issuer)
+                    .map_err(|err| format!("Invalid issuer in attribute scope: {}", err))?;
```
[the remark this answers is not in the document]
Comment (aterga) [reply src/internet_identity_interface/src/internet_identity/types/attributes.rs]: Addressed this

File: src/internet_identity_interface/src/internet_identity/types/attributes.rs [hunk 4]
```diff
@@ -52,6 +54,85 @@ impl std::fmt::Display for AttributeScope {
...
+        problems.push("Empty issuer".to_string());
+    }
+
+    if issuer.len() > OPENID_ISSUER_MAX_LENGTH {
+        problems.push(format!(
+            "Issuer `{}` in attribute scope is too long (max {} chars)",
+            clamp_string_len(issuer, OPENID_ISSUER_MAX_LENGTH),
+            OPENID_ISSUER_MAX_LENGTH
+        ));
+    }
```
[the remark this answers is not in the document]
Comment (aterga) [reply src/internet_identity_interface/src/internet_identity/types/attributes.rs]: Addressed this

File: src/internet_identity_interface/src/internet_identity/types/attributes.rs [hunk 5]
```diff
@@ -338,6 +427,89 @@ mod tests {
...
+                // Edge cases with max_bytes <= 3
+                ("max_bytes 0", "abc", 0, ""),
+                ("max_bytes 1", "abc", 1, "."),
+                ("max_bytes 2", "abc", 2, ".."),
+                ("max_bytes 3", "abc", 3, "abc"),
+                ("max_bytes 3 long string", "abcdef", 3, "..."),
+                // UTF-8 boundary handling (combining diacritics)
+                // y̆ is 'y' (1 byte) + combining breve U+0306 (2 bytes) = 3 bytes total
+                // "y̆zy̆" = 7 bytes total
+                ("utf8 boundary case 1", "y̆zy̆", 4, "y..."), // Only 'y' (1 byte) + '...' fits
```
[the remark this answers is not in the document]
Comment (aterga) [reply src/internet_identity_interface/src/internet_identity/types/attributes.rs:453]: Overkill

File: src/internet_identity_interface/src/internet_identity/types/attributes.rs [hunk 6]
```diff
@@ -338,6 +427,89 @@ mod tests {
...
+            for s in test_strings {
+                for max_bytes in 0..=s.len() + 5 {
+                    let result = ellipsized(s, max_bytes);
+                    // Should always be valid UTF-8
+                    assert!(
+                        result.is_ascii() || std::str::from_utf8(result.as_bytes()).is_ok(),
+                        "ellipsized produced invalid UTF-8 for input: {:?}, max_bytes: {}",
+                        s,
+                        max_bytes
+                    );
```
[the remark this answers is not in the document]
Comment (aterga) [reply src/internet_identity_interface/src/internet_identity/types/attributes.rs:499]: The test also serves as a code-level spec, so this is fine, albeit potentially slightly redundant.

File: src/internet_identity_interface/src/internet_identity/types/attributes.rs [hunk 7]
```diff
@@ -52,6 +54,85 @@ impl std::fmt::Display for AttributeScope {
...
+    if issuer.len() > OPENID_ISSUER_MAX_BYTES {
+        problems.push(format!(
+            "must not exceed {} bytes (got {} bytes)",
+            OPENID_ISSUER_MAX_BYTES,
+            issuer.len(),
+        ));
+    }
+
+    if !issuer.starts_with("https://") {
+        problems.push("must start with `https://`".to_string());
```
[the remark this answers is not in the document]
Comment (aterga) [reply src/internet_identity_interface/src/internet_identity/types/attributes.rs:116]: Overkill
